### ISSUE TITLE
[Question] Do you plan on adding get_profile to get the About section but for pages as well? 

### ISSUE THREAD

**Issue by 97morningstar** [kevinzg/facebook-scraper#332]
I am using this library for a research project and I want to thank you for all your hard work :)

I have a few URLs that I wish to get their profile info (About Section). The problem is that I don't know what type of User the URL belongs to (A page or regular user). Thankfully using your get_posts function I can get the user_id, but if it's a page, I can't use your get_profile for it.

It works fine for User accounts.

The format I am passing to get_posts is:

`"unique name of the page"/posts/"post id"`

Other questions:
1. I am also trying to get the user_id from a post in the following form:

https://www.facebook.com/photo.php?fbid=495386170902530&set=pb.100012934538675.-2207520000..&type=3

But it tells me:

`HTTPError: 404 Client Error: Not Found for url: https://m.facebook.com/photo.php?fbid=495386170902530&set=pb.100012934538675.-2207520000..&type=3/`

I am just using the code from the examples. 

Example of my code:
```

for post in get_posts("elisamartinezfuentes/posts/1190292224745251", cookies=cookie_path, options={"allow_extra_requests": True}):
    account = post['user_id']

print(get_profile(account, cookies=cookie_path))
```

TL;DR: I need to get the info on the public about section for Pages/Groups/Users

I apologize if this is not the correct format for an issue.

**Comment by neon-ninja** [comment]
Yes, I recently added a function called `get_page_info`. Actually, there's no need to resolve account name to user ID, you can pass account names to these functions too.

If you don't know whether it's a page/group/user, you can just run each of the functions and check the output. Here's some sample code:

```python
from facebook_scraper import *

# Profile, Page, Group
accounts = ["elisamartinezfuentes", "Nintendo", "lienminhnongnghieptute"]
functions = [get_profile, get_page_info, get_group_info]
for account in accounts:
    for function in functions:
        try:
            print(function, function(account, cookies="cookies.txt"))
        except:
            pass
```
output:
```python
<function get_profile at 0x7fd8a54f2430> {'Name': 'Elisa Martínez Fuentes', 'Education': 'UMCC Universidad Matanzas Camilo ciefuengos\nIngeniería informática', 'Work': 'Allstate\nTest Engineer Intern\nOn May 18, 2019\nIrving, Texas\nJoyLabs\nWallbreakers Software Engineer Trainee\nIn 2019\nConroe, Texas', 'Places Lived': 'Houston, Texas\nCurrent City\nMoa, Cuba\nHometown', 'Contact Info': {'Facebook': '/elisamartinezfuentes', 'GitHub': '97morningstar', 'Instagram': '97morningstar'}, 'Basic Info': 'August 24, 1997\nBirthday\nFemale\nGender\nEspanol and English\nLanguages', 'Other Names': 'Elisa\nNickname', 'Relationship': {'to': 'Mario Bernal Jr.', 'type': 'Engaged', 'since': 'Since February 7, 2021'}, 'Family Members': 'Mi hermana\nSister\nMi papa\nFather\nMi mama\nMother\nIsmaray Martinez\nFamily member', 'Life Events': '2021\nGot Engaged to Mario Bernal Jr.\n2019\nIn a Relationship with Mario Bernal Jr.'}
<function get_page_info at 0x7fd8a54f24c0> {'name': 'Elisa Martínez Fuentes', 'identifier': 100012934538675, 'url': 'https://www.facebook.com/elisamartinezfuentes', 'image': 'https://scontent.fhlz2-1.fna.fbcdn.net/v/t1.6435-1/fr/cp0/e15/q65/74634432_755065824934562_3872097589268578304_n.jpg?_nc_cat=111&ccb=1-3&_nc_sid=8d0d6e&efg=eyJpIjoidCJ9&_nc_ohc=ctTSociQCz8AX9UVjqr&_nc_ht=scontent.fhlz2-1.fna&tp=14&oh=60191d22c4524cf806a9dae988c883ea&oe=60E96F7A', 'sameAs': None, 'dateCreated': None, 'type': 'Person'}
<function get_profile at 0x7fd8a54f2430> {'Name': 'Nintendo - About'}
<function get_page_info at 0x7fd8a54f24c0> {'name': 'Nintendo', 'identifier': 119240841493711, 'url': 'https://www.facebook.com/Nintendo/', 'image': 'https://scontent.fhlz2-1.fna.fbcdn.net/v/t1.18169-1/13138816_1025939437490509_1827627750936743506_n.png?_nc_cat=1&ccb=1-3&_nc_sid=8d0d6e&efg=eyJpIjoidCJ9&_nc_ohc=EVVea4o3SQ4AX_FxXMq&_nc_ht=scontent.fhlz2-1.fna&oh=4acd1086d0e2854ab8e31efff3ba4c22&oe=60EA12D2', 'sameAs': 'http://www.nintendo.com', 'dateCreated': '2011-06-01T15:23:50-0700', 'type': 'Organization', 'likes': 5412964, 'followers': 5471921}
<function get_group_info at 0x7fd8a54f2550> {'id': '627551287825967', 'name': 'LIÊN MINH NÔNG NGHIỆP TỬ TẾ', 'type': 'Public group', 'members': [{'name': 'Hoàng Nguyen', 'link': '/profile.php?id=100014268012641'}, {'name': 'Nguyễn Quốc Nam', 'link': '/profile.php?id=100015991331351'}, {'name': 'Nguyễn Thị Thu', 'link': '/thu.tqd'}, {'name': 'Thiên Vũ', 'link': '/profile.php?id=100027172146614'}, {'name': 'Trần Huế', 'link': '/profile.php?id=100025847746706'}, {'name': 'Ban Icat', 'link': '/ban.nguyenban.3'}, {'name': 'Dương Ngọc Phúc', 'link': '/profile.php?id=100007471802699'}, {'name': 'Dương Đình Tường', 'link': '/profile.php?id=100008623893421'}, {'name': 'Hà Trần Thị Thuý', 'link': '/ha.tranthithuy.1'}, {'name': 'Liên Giun Quế Ght', 'link': '/nguyenthi.lien.5'}, {'name': 'Lê Tuyền', 'link': '/lethingoctuyen1989'}, {'name': 'Nguyễn Luyến', 'link': '/profile.php?id=100008158538378'}, {'name': 'Nguyễn Phúc', 'link': '/profile.php?id=100026866214726'}, {'name': 'Phương Nguyễn Quang', 'link': '/phuong.nguyenquang.7'}, {'name': 'Quang Thai Tran', 'link': '/thaianh10001'}, {'name': 'Thu Hà', 'link': '/thuha2190'}, {'name': 'Tung Tran', 'link': '/tung.tran.33046'}, {'name': 'Tài MU', 'link': '/taimu1503'}, {'name': 'Đoàn Thanh Bình', 'link': '/profile.php?id=100003338407049'}], 'admins': [{'name': 'Hoàng Công', 'link': '/hoang.cong.79'}, {'name': 'Hoàng Nguyen', 'link': '/profile.php'}, {'name': 'Hoàng Thanh', 'link': '/hoangthanh588'}, {'name': 'Hưng Rivers East', 'link': '/HungRiversEast'}]}
```
Some functions do work with input that doesn't match their intended use, as you can see. But you can filter by expected keys.

**Comment by 97morningstar** [comment]
Thank you @neon-ninja, python is amazing! :) 

What about the "About" section on the Fb pages? I think with the get_profile and get_group_info I have almost everything I need. Do you plan on adding that in the future? If not I can try to add it and help this project. 

(I will check my problem with "photo.php" later :), I will open a new one with more clarity/testing on that).

These are the types of URL I am working with, and trying to get their profile/about info from: 

https://gist.github.com/97morningstar/9942fbae0e65c26c37d32d975b29bec3

With your little code snippet, I can start classifying users, groups, and pages from any Fb URL given (hopefully). Thanks!

**Comment by neon-ninja** [comment]
What information are you trying to extract from a page's about section that `get_page_info` doesn't extract? Do you have a sample page where it doesn't work?

With directly extracting a post, you should signify to the scraper that the URL represents one post so that it doesn't try to paginate. The argument for this is post_urls.

```python
get_posts(post_urls=["https://www.facebook.com/photo.php?fbid=495386170902530&set=pb.100012934538675.-2207520000..&type=3"], cookies="cookies.txt")
```

**Comment by neon-ninja** [comment]
https://github.com/kevinzg/facebook-scraper/commit/5c904e8761735003a93c2e3c9803aa4259ee088a might help

**Comment by 97morningstar** [comment]
`What information are you trying to extract from a page's about section that get_page_info doesn't extract?`

I am trying to access the info on this page, for example:

https://www.facebook.com/cityofgalveston/about/?ref=page_internal

The General, Hours, Additional Contact Info, and More Info sections, if they exist. Also website, phone, type of organization, of course, as long as they are all of public access.

Thank you again, I uninstalled and installed again the package using pip. Should I download the repo here instead?

`
 Do you have a sample page where it doesn't work?`

Yes, this is my code snippet and output:

```
for post in get_posts("cityofgalveston/posts/4199446900116960", cookies=cookie_path):
    account = post['user_id']

print(get_page_info(account, cookies=cookie_path))
```

output:

`{'name': 'City of Galveston, Texas - Government', 'identifier': 109633465765011, 'url': 'https://www.facebook.com/cityofgalveston/', 'image': 'https://scontent-dfw5-1.xx.fbcdn.net/v/t1.18169-1/fr/cp0/e15/q65/10929550_830039863724364_7151476885763355021_n.jpg?_nc_cat=110&ccb=1-3&_nc_sid=8d0d6e&efg=eyJpIjoidCJ9&_nc_ohc=4judlR7PZosAX93bLqE&_nc_ht=scontent-dfw5-1.xx&tp=14&oh=aa08ac379760b6c34d803ce623057564&oe=60C801D7', 'sameAs': 'http://www.galvestontx.gov', 'dateCreated': '2010-09-17T17:11:41-0700', 'type': 'Organization', 'followers': 47581}`

I can't state enough my gratitude for this package, I spent almost a week trying the Facebook Graph API, until I realized that they need to approve every permission, and you may not get approved. 

Note: This project is part of a Research Experience of Undergraduates program I am in.

**Comment by neon-ninja** [comment]
I haven't created a PyPI release containing this commit yet, you can use `​pip install git+https://github.com/kevinzg/facebook-scraper.git` to install the latest master branch

**Comment by neon-ninja** [comment]
```python
from facebook_scraper import *
from pprint import pprint
pprint(get_page_info("cityofgalveston", cookies="cookies.txt"))
```
outputs

```python
{'about': '823 Rosenberg, Galveston, TX 77550\n'
          'Get Directions\n'
          'Closed Now\n'
          '·\n'
          '8 AM - 5 PM\n'
          'Closed Now\n'
          '·\n'
          '8 AM - 5 PM\n'
          'Sunday\n'
          'Monday\n'
          'Tuesday\n'
          'Wednesday\n'
          'Thursday\n'
          'Friday\n'
          'Saturday\n'
          'CLOSED\n'
          '8 AM - 5 PM\n'
          '8 AM - 5 PM\n'
          '8 AM - 5 PM\n'
          '8 AM - 5 PM\n'
          '8 AM - 5 PM\n'
          'CLOSED\n'
          '+1 409-797-3500\n'
          'http://www.galvestontx.gov/\n'
          "Welcome to the City of Galveston's official Facebook Page!\n"
          'Government Organization · Public & Government Service\n'
          'Send Message\n'
          'www.galvestontx.gov',
 'dateCreated': '2010-09-17T17:11:41-0700',
 'followers': 47591,
 'identifier': 109633465765011,
 'image': 'https://scontent.fakl1-3.fna.fbcdn.net/v/t1.18169-1/fr/cp0/e15/q65/10929550_830039863724364_7151476885763355021_n.jpg?_nc_cat=110&ccb=1-3&_nc_sid=8d0d6e&efg=eyJpIjoidCJ9&_nc_ohc=GrtCkObn3ToAX-zqxaf&_nc_ht=scontent.fakl1-3.fna&tp=14&oh=14601e305ca41eeb450c6a3434e6eb22&oe=60CBF657',
 'likes': 44080,
 'name': 'City of Galveston, Texas - Government',
 'sameAs': 'http://www.galvestontx.gov',
 'type': 'Organization',
 'url': 'https://www.facebook.com/cityofgalveston/'}
```

**Comment by 97morningstar** [comment]
Thank you so much @neon-ninja  
I am sorry to bother you again, I ran the code again with the updated code now and got this error when calling get_page_info

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-13-3c48354637ff> in <module>
     14 
     15 
---> 16 print(get_page_info("cityofgalveston", cookies=cookie_path))
     17 
     18 #post_urls = ["https://www.facebook.com/photo.php?fbid=495386170902530&set=pb.100012934538675.-2207520000..&type=3"]

~\anaconda3\anaconda\envs\findm\lib\site-packages\facebook_scraper\__init__.py in get_page_info(account, **kwargs)
     77     cookies = kwargs.pop('cookies', None)
     78     set_cookies(cookies)
---> 79     return _scraper.get_page_info(account, **kwargs)
     80 
     81 

~\anaconda3\anaconda\envs\findm\lib\site-packages\facebook_scraper\facebook_scraper.py in get_page_info(self, page, **kwargs)
    230             return None
    231         meta = json.loads(elem.text)
--> 232         result = meta["author"]
    233         result["type"] = result.pop("@type")
    234         desc = resp.html.find("meta[name='description']", first=True)

KeyError: 'author'
```

**Comment by neon-ninja** [comment]
Try this - https://github.com/kevinzg/facebook-scraper/commit/433d54810318cf43f0111c6afdcc0989f5433b6a
